pallets/api/src/messaging/mod.rs

```diff
@@ -450,7 +503,7 @@ pub mod pallet {
 			let querier_location = T::OriginConverter::try_convert(origin.clone())
 				.map_err(|_| Error::<T>::OriginConversionFailed)?;
 			let origin = ensure_signed(origin)?;
-			ensure!(!Messages::<T>::contains_key(&origin, &id), Error::<T>::MessageExists);
+			ensure!(!Messages::<T>::contains_key(&origin, id), Error::<T>::MessageExists);
```

Comment:
curious why you removed the borrow (you did the same further up as well)? In WASM (runtime), references are generally more efficient to be used

Reply:
i believe rust happily type coerces the value to the borrowed form, since its a [u8;32]
This removes the indirection required for a pointer